extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -168,6 +169,21 @@ private String guessDriver(Optional<JdbcDataSourceBuildItem> jdbcDataSource) {
                     .add(new ReflectiveClassBuildItem(true, true, "io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob"));
             reflectiveClasses
                     .add(new ReflectiveClassBuildItem(true, false, QuarkusQuartzConnectionPoolProvider.class.getName()));
+        } else if (config.storeType.isCustom()) {
+            Optional<QuartzExtensionPointConfig> customStore = config.customStore();
+            if (!customStore.isPresent()) {
+                throw new ConfigurationException("Custom job store class is not configured properly");
+            }
+            try {
+                if (!JobStore.class.isAssignableFrom(
+                        Class.forName(customStore.get().clazz, false, Thread.currentThread().getContextClassLoader()))) {
```

Comment:
I'm not quite sure it's always safe to load the impl class at build time. It might be better to find the class in the Jandex index and verify that it implements the `JobStore`.

Reply:
Add BeanArchiveIndexBuildItem, write code

 catch (ClassNotFoundException e) {
...
}

```java
  List<ReflectiveClassBuildItem> reflectiveClasses(BeanArchiveIndexBuildItem beanArchiveIndex, QuartzBuildTimeConfig config,
            QuartzJDBCDriverDialectBuildItem driverDialect) {
...
 Optional<QuartzExtensionPointConfig> customStore = config.customStore();
            if (!customStore.isPresent()) {
                throw new ConfigurationException("Custom job store class is not configured properly");
            }
            try {
                if (!JobStore.class.isAssignableFrom(
                        Class.forName(customStore.get().clazz, false, Thread.currentThread().getContextClassLoader()))) {
                    throw new ConfigurationException(
                            String.format(
                                    "The provided class \"%s\" does not exist. Make sure it exists and it implements \"%s\"",
                                    customStore.get().clazz, JobStore.class.getName()));
                }
            } catch (ClassNotFoundException e) {
                boolean isAssignableFrom = false;
                ClassInfo classInfo = beanArchiveIndex.getIndex().getClassByName(DotName.createSimple(customStore.get().clazz));
                if (classInfo != null) {
                    DotName jobStoreDotName = DotName.createSimple(JobStore.class.getName());
                    for (DotName interfaceName : classInfo.interfaceNames()) {
                        if (interfaceName.equals(jobStoreDotName)) {
                            isAssignableFrom = true;
                            break;
                        }
                    }
                }
                if (!isAssignableFrom) {
                    throw new ConfigurationException(
                            String.format(
                                    "The provided class \"%s\" does not exist. Make sure it exists and it implements \"%s\"",
                                    customStore.get().clazz, JobStore.class.getName()));
                }
            }
            reflectiveClasses.add(new ReflectiveClassBuildItem(true, true, customStore.get().clazz));
```

Reply:
Hi @mkouba would you have a better suggestion here? I do not think a bean archive index is what should be used. Maybe the `ApplicationArchivesBuildItem` is the most appropriate here.

I also think we should attempt to load the class directly from Jandex and not attempt a try catch like what's current done.

Reply:
@mkouba I actually have a question, the Job.class will mostly come from external libraries. Will they be automatically index or the user will have to make sure that they are indexed following https://quarkus.io/guides/all-config#quarkus-core_quarkus.index-dependency.index-dependency-artifacts-on-the-classpath-that-should-also-be-indexed?

Reply:
I think that `io.quarkus.deployment.builditem.CombinedIndexBuildItem.getComputingIndex()` should work in most cases. However, I don't think we should try to load the class first and then use the index as a fallback if `ClassNotFoundException` is thrown, but use the index solely instead.

Reply:
@leotu Pls update the PR instead so that we can review the code in context (and the snippet above looks good ;-)